phoenix-core-server/src/main/java/org/apache/phoenix/replication/ReplicationLogTracker.java

```diff
@@ -0,0 +1,518 @@
+        }
+        return Optional.of(parts[parts.length - 1].split("\\.")[0]);
+    }
+
+    /**
+     * Extracts everything except the UUID (last part) from a file path.
+     * For example, from "1704153600000_rs1_12345678-1234-1234-1234-123456789abc.plog"
+     * This method will return "1704153600000_rs1"
+     * @param file - The file path to extract prefix from.
+     */
+    protected String getFilePrefix(Path file) {
+        String fileName = file.getName();
+        String[] parts = fileName.split("_");
+        if (parts.length < 3) {
+            return fileName.split("\\.")[0]; // Return full filename if no underscore found
+        }
+
+        // Return everything except the last part (UUID)
+        StringBuilder prefix = new StringBuilder();
+        for (int i = 0; i < parts.length - 1; i++) {
+            if (i > 0) {
+                prefix.append("_");
+            }
+            prefix.append(parts[i]);
+        }
+
+        return prefix.toString();
+    }
+
+    /**
+     * No op implementation for marking a file as failed as its expected to be picked by
+     * by other process from in-progress directory
+     * @param file - The file which needs to be marked as failed
+     * @return - true if marked as failed, false otherwise
+     */
+    public boolean markFailed(final Path file) {
+        getMetrics().incrementMarkFileFailedRequestCount();
+        return true;
+    }
+
+    public FileSystem getFileSystem() {
+        return this.fileSystem;
+    }
+
+    public ReplicationShardDirectoryManager getReplicationShardDirectoryManager() {
+        return this.replicationShardDirectoryManager;
+    }
+
+    protected String getHaGroupName() {
+        return this.haGroupName;
+    }
+
+    protected Configuration getConf() {
+        return this.conf;
+    }
+
+    protected Path getInProgressDirPath() {
+        return this.inProgressDirPath;
+    }
+
+    protected MetricsReplicationLogTracker getMetrics() {
+        return this.metrics;
+    }
+
+    /**
+     * Creates a directory if it doesn't exist.
+     */
+    private void createDirectoryIfNotExists(Path directoryPath) throws IOException {
+        if (!fileSystem.exists(directoryPath)) {
+            LOG.info("Creating directory {}", directoryPath);
+            if (!fileSystem.mkdirs(directoryPath)) {
+                throw new IOException("Failed to create directory: " + directoryPath);
+            }
+        }
+    }
+
+    /**
+     * Enum representing the type of replication log directory.
+     * IN: Directory created on standby cluster for Incoming replication log files
+     * OUT: Directory created on primary cluster for Outgoing replication log files
+     */
+    public enum DirectoryType {
+        IN("in"),
+        OUT("out");
+
+        private final String name;
+
+        DirectoryType(final String name) {
+            this.name = name;
+        }
+
+        public String getName() {
+            return this.name;
+        }
+    }
```

Comment:
I think we should get rid of this enum. It is not really an enum. What if we use a different terminology. These are just string constants but they don't need to be defined here but at a higher level. The ReplicationLogTracker doesn't need to know about it. It should take a path as input and just work with it.